miso-web/src/main/webapp/scripts/list_orderaliquot.js

```diff
@@ -0,0 +1,149 @@
+            var dupes = [];
+            PoolOrder.getAliquots().forEach(function(orderAli) {
+              if (aliquots.map(Utils.array.getId).indexOf(orderAli.aliquot.id) !== -1) {
+                dupes.push(orderAli.aliquot);
+              }
+            });
+            if (dupes.length) {
+              Utils.showOkDialog('Error', ['The following aliquots are already included in this order:'].concat(dupes
+                  .map(function(aliquot) {
+                    return '* ' + aliquot.name + ' (' + aliquot.alias + ')';
+                  })));
+            } else {
+              Utils.showDialog('Edit Proportions', 'Add', aliquots.map(function(aliquot) {
+                return {
+                  label: aliquot.name + ' (' + aliquot.alias + ')',
+                  type: 'int',
+                  property: 'aliquot' + aliquot.id + 'Proportion',
+                  required: true,
+                  value: 1
+                };
+              }), function(proportionResults) {
+                PoolOrder.addAliquots(aliquots.map(function(aliquot) {
+                  return {
+                    id: aliquot.id,
+                    aliquot: aliquot,
+                    proportion: proportionResults['aliquot' + aliquot.id + 'Proportion']
+                  };
+                }));
+              })
+            }
+          });
+        });
+      }
+    }];
+  },
+  createColumns: function(config, projectId) {
+    return [{
+      'sTitle': 'Name',
+      'mData': 'aliquot.id', // for sorting purposes
+      'include': true,
+      'iSortPriority': 1,
+      "mRender": function(data, type, full) {
+        if (type === 'display') {
+          return '<a href="' + Urls.ui.libraryAliquots.edit(data) + '">' + full.aliquot.name + '</a>';
+        }
+        return data;
+      }
+    }, ListUtils.labelHyperlinkColumn("Alias", Urls.ui.libraryAliquots.edit, function(orderAliquot) {
+      return orderAliquot.aliquot.id;
+    }, "aliquot.alias", 0, true), {
+      'sTitle': 'Proportion',
+      'sType': 'numeric',
+      'mData': 'proportion',
+      'include': !config.add,
+      'iSortPriority': 0
+    }, ListUtils.idHyperlinkColumn("Sample Name", Urls.ui.samples.edit, "aliquot.sampleId", function(orderAli) {
+      return orderAli.aliquot.sampleId;
+    }, 0, true, "noPrint"), ListUtils.labelHyperlinkColumn("Sample Alias", Urls.ui.samples.edit, function(orderAli) {
+      return orderAli.aliquot.sampleId;
+    }, "aliquot.sampleAlias", 0, true), {
+      'sTitle': 'Indices',
+      'mData': 'aliquot.indexIds',
+      'include': true,
+      'bSortable': false,
+      'iSortPriority': 0,
+      'mRender': function(data, type, full) {
+        var indices = Constants.indexFamilies.reduce(function(acc, family) {
+          return acc.concat(family.indices.filter(function(index) {
+            return data.indexOf(index.id) != -1;
```

Comment:
If you try to add a library aliquot from a library that has no indices, it will explode here and the jQuery table locks up. Maybe there should be a filter at the previous step that disallows any indexless entries to be added.

Reply:
It should be possible to create a pool order for a single aliquot with no indices so I've just changed it to show "(none)" instead. There is a separate ticket for disallowing index collisions in a pool order, which should also prevent multiple libraries with missing indices